samcli/lib/build/app_builder.py

```diff
@@ -663,3 +706,89 @@ def _parse_builder_response(stdout_data: str, image_name: str) -> Dict:
             raise ValueError(msg)
 
         return cast(Dict, response)
+
+    @staticmethod
+    def _make_env_vars(function: Function, file_env_vars: Dict, inline_env_vars: Optional[Dict]) -> Dict:
+        """Returns the environment variables configuration for this function
+
+        Parameters
+        ----------
+        function : samcli.lib.providers.provider.Function
+            Lambda function to generate the configuration for
+
+        Returns
+        -------
+        dictionary
+            Environment variable configuration for this function
+
+        Raises
+        ------
+        samcli.commands.local.lib.exceptions.OverridesNotWellDefinedError
+            If the environment dict is in the wrong format to process environment vars
+
+        """
+
+        name = function.name
+        result = {}
+
+        if file_env_vars and inline_env_vars:
+            for file_env_var in file_env_vars.values():
+                if not isinstance(file_env_var, dict):
+                    reason = """
+                                Environment variables in incorrect format
+                                """
+                    LOG.debug(reason)
+                    raise OverridesNotWellDefinedError(reason)
+
+            for inline_env_var in inline_env_vars.values():
+                if not isinstance(inline_env_var, dict):
+                    reason = """
+                                Environment variables in incorrect format
+                                """
+                    LOG.debug(reason)
+                    raise OverridesNotWellDefinedError(reason)
+
+            LOG.debug("Environment variables data is being parsed")
+            # CloudFormation parameter file format
+            parameter_result = file_env_vars.get("Parameters", {})
+            result = parameter_result.copy()
+            inline_parameter_result = inline_env_vars.get("Parameters", {})
+            result.update(inline_parameter_result)
+            specific_result = file_env_vars.get(name, {})
+            result.update(specific_result)
+            inline_specific_result = inline_env_vars.get(name, {})
+            result.update(inline_specific_result)
```

Comment:
question: for `inline_parameter_result` and `inline_specific_result`, who has higher precedence? Maybe worth to drop a note here.

Reply:
I am putting a comment for this function to specify the precedence